### ISSUE TITLE
try-except-raise does not consider subclasses

### ISSUE THREAD

**Issue by kchen** [PyCQA/pylint#2284]
### Steps to reproduce
Run pylint on the following file:
```
"""try-except-raise test module"""

class AAAException(Exception):
    """AAAException"""
    pass

class BBBException(AAAException):
    """BBBException"""
    pass

def ccc():
    """try-except-raise test function"""

    try:
        raise BBBException("asdf")
    except BBBException:
        raise
    except AAAException:
        raise BBBException("raised from AAAException")

ccc()
```

### Current behavior
************* Module try_except_raise
try_except_raise.py:16:4: E0705: The except handler raises immediately (try-except-raise)

### Expected behavior
No issue is reported because BBBException is caught and raised immediately to avoid being caught and handled by the parent exception class's handler.

### pylint --version output
pylint 2.0.0.dev2
astroid 2.0.0.dev4
Python 3.7.0 (default, Jul 13 2018, 22:37:47) 
[GCC 5.4.0 20160609]

**Comment by PCManticore** [comment]
This is odd, but valid. Thanks for the report!

**Comment by RemiCardona** [comment]
Got this one too.

**Comment by PCManticore** [comment]
This is fixed now.
